src/extension.ts

```diff
@@ -61,3 +65,25 @@ function addExtensionChangeListener(context: ExtensionContext): void {
         context.subscriptions.push(extensionChangeListener);
     }
 }
+
+async function handleDeprecatedTasks(): Promise<void> {
+    await contextManager.setContextValue(Context.SHOW_DEPRECATED_TASKS, true);
+    const deprecatedTasks = await tasks.fetchTasks({ type: "java" });
+    if (deprecatedTasks?.length) {
+        let tasksJsonPath;
+        const options = [];
+        if (workspace.workspaceFolders?.length === 1) {
+            tasksJsonPath = path.join(workspace.workspaceFolders[0].uri.fsPath, ".vscode", "tasks.json");
+            if (await fse.pathExists(tasksJsonPath)) {
+                options.push("Open tasks.json");
```

Comment:
I prefer to providing an option that directly update the task type (if we can find the correct deprecated tasks).